src/main/java/com/epam/lab/paymentsystem/service/impl/CardServiceImpl.java

```diff
@@ -94,18 +95,56 @@ public Card createCard(CardDto card) throws UnsupportedOperationException {
     LOGGER.info("Creating new card");
     Account account = accountService.getAccountById(card.getAccountId());
     User user = userService.getUserByLogin(card.getUserLogin());
+    String cardNumber = generateCardNumber();
```

Comment:
There is no need to create cardNumber if user is null, for example.
Please, put this method after the user check